python/lsst/daf/butler/registry/datasets/byDimensions/_manager.py

```diff
@@ -115,6 +115,22 @@ def refresh(self, *, universe: DimensionUniverse) -> None:
         self._byName = byName
         self._byId = byId
 
+    def deleteDatasetType(self, name: str) -> None:
+        # Docstring inherited from DatasetRecordStorageManager.
+        compositeName, componentName = DatasetType.splitDatasetTypeName(name)
+        if componentName is not None:
+            raise ValueError(f"Cannot delete a dataset type of a component of a composite (given {name})")
+        storage = self._byName.get(compositeName)
+        if storage is None:
+            raise KeyError(f"DatasetType {name} is not registered.")
+
+        # Delete the row
+        self._db.delete(self._static.dataset_type, ["name"], {"name": name})
+
+        # Now delete the cache entries
```

Comment:
Should I pass in a universe and call `refresh` here instead?

Reply:
I think this is safe enough, but dataset type deletion ought to be rare enough the cost of `refresh` shouldn't matter, I think.

Reply:
Actually, with the refresh call here and the delete being a no-op if it doesn't exist, I can remove the lines above that check for existence in the cache so this all gets much simpler.